Type errors due to assuming DOM libraries in Node projects

I upgraded TypeDoc to use the latest Shiki, and started seeing this error, since it does not include "DOM" in its lib and skipLibCheck is off.

> Error: node_modules/shiki/dist/index.d.ts(316,55): error TS2304: Cannot find name 'Response'.

--- a/packages/renderer-svg/src/measureMonospaceTypeface.ts
+++ b/packages/renderer-svg/src/measureMonospaceTypeface.ts
@@ -1,3 +1,5 @@
+/// <reference path="./global.d.ts" />
+
 function measureFont([fontName, fontSize]: [string, number]) {
   /**
    * Measure `M` for width

--- a/packages/shiki/src/loader.ts
+++ b/packages/shiki/src/loader.ts
@@ -4,6 +4,20 @@ import { loadWASM, createOnigScanner, createOnigString } from 'vscode-oniguruma'
 import { parse, ParseError } from 'jsonc-parser'
 import type { IShikiTheme } from './types'
 
+// These will not be defined for Node which don't have "DOM" in their lib.
+// So declare the minimal interface we need.
+declare global {
+  interface Window {
+    WorkerGlobalScope: any
+  }
+  var self: Window & typeof globalThis
+  function fetch(url: string): Promise<Response>
+  interface Response {
+    json(): Promise<any>
+    text(): Promise<any>
+  }
+}
+
 export const isWebWorker =
   typeof self !== 'undefined' && typeof self.WorkerGlobalScope !== 'undefined'
 export const isNode =

--- a/packages/shiki/tsconfig.json
+++ b/packages/shiki/tsconfig.json
@@ -4,7 +4,7 @@
     "target": "ES2020",
     "esModuleInterop": true,
     "moduleResolution": "node",
-    "lib": ["esnext", "DOM", "WebWorker"],
+    "lib": ["ESNext"],
     "sourceMap": true
   },
   "exclude": ["samples"]

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,12 +1,9 @@
 {
-  "compilerOptions": {
-    "module": "commonjs",
-    "target": "ES2020",
-    "declaration": true,
-    "esModuleInterop": true,
-    "moduleResolution": "node",
-    "lib": ["es2015", "es2016.array.include", "DOM", "WebWorker"],
-    "sourceMap": true
-  },
-  "references": [{ "path": "packages/shiki" }, { "path": "packages/renderer-svg" }]
+  "files": [],
+  "references": [
+    { "path": "packages/shiki" },
+    { "path": "packages/renderer-svg" },
+    { "path": "packages/renderer-path" },
+    { "path": "packages/vuepress-plugin" }
+  ]
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,9 +1,11 @@
 {
+  "compilerOptions": {
+    "esModuleInterop": true
+  },
   "files": [],
   "references": [
     { "path": "packages/shiki" },
     { "path": "packages/renderer-svg" },
-    { "path": "packages/renderer-path" },
-    { "path": "packages/vuepress-plugin" }
+    { "path": "packages/renderer-path" }
   ]
 }

